core/src/main/scala/cats/Monad.scala

```diff
@@ -162,6 +162,20 @@ trait Monad[F[_]] extends FlatMap[F] with Applicative[F] {
     tailRecM(branches.toList)(step)
   }
 
+  /**
+   * Returns the given argument (mapped to Unit) if `cond` evaluates to `false`, otherwise,
+   * unit lifted into F.
+   */
+  def unlessM[A](f: F[A])(cond: F[Boolean]): F[Unit] =
+    flatMap(cond)(bool => if (bool) unit else void(f))
+
+  /**
+   * Returns the given argument (mapped to Unit) if `cond` evaluates to `true`, otherwise,
+   * unit lifted into F.
+   */
+  def whenM[A](f: F[A])(cond: F[Boolean]): F[Unit] =
+    flatMap(cond)(bool => if (bool) void(f) else unit)
```

Comment:
Just to promote Mouse, you may be interested to glance at this syntax
https://github.com/typelevel/mouse/blob/main/shared/src/main/scala/mouse/boolean.scala#L73
It covers `whenA` and `unlessA` for boolean values.